lib/handlers/connect.js

```diff
@@ -63,10 +63,26 @@ function authenticate (arg, done) {
       client.broker.registerClient(client)
       return done()
     } else if (err) {
-      write(client, {
-        cmd: 'connack',
-        returnCode: 4
-      }, client.close.bind(client, done))
+      if (err && !err.returnCode) {
+        write(client, {
+          cmd: 'connack',
+          returnCode: 4
+        }, client.close.bind(client, done))
+        return
+      } else {
```

Comment:
That was for checking if somebody doesn't send the returnCode property. I can merge it with the if below though.